sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -439,6 +439,12 @@ object HiveTypeCoercion {
             DecimalType(min(p1 - s1, p2 - s2) + max(s1, s2), max(s1, s2))
           )
 
+        case Pmod(e1 @ DecimalType.Expression(p1, s1), e2 @ DecimalType.Expression(p2, s2)) =>
```

Comment:
I guess the casting rule for `pmod` is not a general one, just as like the Remainder  `case Remainder.... DecimalType(min(p1 - s1, p2 - s2) + max(s1, s2), max(s1, s2))` is the special casting rule for the result in `HiveTypeCoercion`.